### PR TITLE
Changelog for upcoming 3.1.0 release

### PR DESCRIPTION
**Pull request by cben**
Sunday release date preliminary, will adjust if needed and merge when we do actual release.

I'm holding back #323 because it's not fully backwards compatible and I believe will need a 4.0.0 release.

cc @moolitayer @Stono @lambertjosh @pravi

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,21 @@
...
+### Fixed
+- Fixed watch `.finish` sometimes caused `HTTP::ConnectionError` exception from the reading loop (#315).
+
+### Added
+- `get_pod_log` now has `timestamps`, `since_time` (#319) and `tail_lines` (#326) params.
+- `Kubeclient::Config::Context#namespace` now set, if present in kubeconfig file (#308).
+- Improved README directions for authenticating within a kubernetes cluster (#316).
+- `Kubeclient::GoogleApplicationDefaultCredentials` helper for Google application default credentials (#213). Needs `googleauth` gem.
+- New `as: :parsed` and `as: :parsed_symbolized` formats (#306).
+- Allow setting default `as:` format for whole client (#299, #305).
```
Comment (moolitayer) [inline CHANGELOG.md]: for the whole client

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,21 @@
...
+- Relaxed `recursive-open-struct` to allow 1.1+ as well (#313).
```
Comment (moolitayer) [inline CHANGELOG.md]: Relaxed `recursive-open-struct` dependency

**Comment by cben** [comment]
done, PTAL

**Comment by cben** [comment]
Released 3.1.0.

**Pull request merged**